haystack/utils/url_validation.py

```diff
@@ -0,0 +1,6 @@
+from urllib.parse import urlparse
+
+
+def is_valid_http_url(url) -> bool:
+    r = urlparse(url)
+    return all([r.scheme in ["http", "https"], r.netloc])
```

Comment:
This can be ignored.
I introduced and tested it in #7464, but I also added it in this PR to make the 2 PRs independent.